src/envparse.cpp

```diff
@@ -0,0 +1,275 @@
+/**
+ * @file envparse.cpp
+ * @author Derek Tan
+ * @brief Implements parser for environment secret files with simple .env syntax .
+ * @note Does not support boolean literals!
+ * @todo Refactor lexing code to accept character -> bool lambdas as the token checking predicates.
+ * @date 2023-12-12
+ */
+
+#include "parser/envparse.hpp"
+
+/* Lexer Helpers */
+
+[[nodiscard]] constexpr bool parser::utils::is_whitespace(char c)
+{
+    return c == ' ' || c == '\r' || c == '\n' || c == '\t';
+}
+
+[[nodiscard]] constexpr bool parser::utils::is_alpha(char c)
+{
+    return (c >= 'A' && c <= 'Z') || (c >= 'a' || c <= 'z') || c == '_';
+}
+
+[[nodiscard]] constexpr bool parser::utils::is_digit(char c)
+{
+    return c >= '0' && c <= '9';
+}
+
+/* Helper Impl. */
+
+[[nodiscard]] bool parser::token_as_text(std::string& buffer_str, const char* source, const Token& token)
+{
+    if (token.length == 0)
+    {
+        return false;
+    }
+
+    buffer_str.clear();
+
+    size_t begin = static_cast<size_t>(token.begin);
+    size_t end = static_cast<size_t>(token.begin + token.length - 1UL);
+
+    for (size_t source_i = begin; source_i <= end; source_i++)
+    {
+        buffer_str += source[source_i];
+    }
+
+    return true;
+}
+
+/* Lexer Impl. */
+
+parser::Lexer::Lexer(const char* source_ptr)
+{
+    if (!source_ptr)
+    {
+        throw std::invalid_argument("Invalid handle to source, found nullptr.");
+    }
+
+    this->source_ptr = source_ptr;
+    this->source_length = std::strlen(source_ptr);
+    this->source_pos = 0U;
+    this->source_line = 1U;
+}
+
+constexpr bool parser::Lexer::has_ended() const
+{
+    return this->source_pos >= this->source_length;
+}
+
+constexpr uint32_t parser::Lexer::tell_line() const
+{
+    return this->source_line;
+}
+
+void parser::Lexer::skip_comment()
+{
+    char temp = '\0';
+
+    while (this->source_pos < this->source_length)
+    {
+        temp = source_ptr[this->source_pos];
+
+        if (temp == '\n')
```

Comment:
This is another case where a new line can be scanned by the lexer. Thus, there is a missing line count increment here! 👀

Reply:
The micro fix is now placed. No other newline checks exist to need an line counter increment.